Mono.Debugging.Soft/SoftDebuggerSession.cs

```diff
@@ -1947,10 +1947,11 @@ void HandleBreakEventSet (Event[] es, bool dequeuing)
 						}
 						
 						if (hasBreakInfo) {
-							if (currentStepRequest != null &&
-							    currentStepRequest.Depth != StepDepth.Out &&
-							    binfo.Location.ILOffset == currentAddress && 
-							    e.Thread.Id == currentStepRequest.Thread.Id &&
+							if (currentRequest != null &&
+								currentRequest is StepEventRequest &&
+								(currentRequest as StepEventRequest).Depth != StepDepth.Out &&
```

Comment:
This could probably be expressed like this:

```csharp
if ((currentRequest as StepEventRequest currentStepRequest) && currentStepRequest.Depth != StepDepth.Out &&
```

Reply:
Done